### PR TITLE
Update PR template

### PR DESCRIPTION
**Pull request by spydon**
# Description

Updated the PR template to reflect removal of the develop branch

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

# Checklist:

If something is unclear, please submit the PR anyways and ask about what you thought was unclear.

- [x] This branch is based on `develop`
- [x] This PR is targeted to merge into `develop`
- [x] I have added an entry under `[next]` in `CHANGELOG.md`
- [x] I have formatted my code with `flutter format`
- [x] I have made corresponding changes to the documentation
- [x] I have added examples for new features in `doc/examples`
- [x] The continuous integration (CI) is passing

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -13,12 +13,11 @@ Please delete options that are not relevant.
...
 - [ ] This change requires a documentation update
 
-# Checklist:
+## Checklist:
 
 If something is unclear, please submit the PR anyways and ask about what you thought was unclear.
 
-- [ ] This branch is based on `v1.0.0`
-- [ ] This PR is targeted to merge into `v1.0.0` (not `master` or `develop`)
+- [ ] This branch is based on the latest `master`
```
Comment (erickzanardo) [inline .github/pull_request_template.md:20]: I am really confused, aren't we keeping develop until we have the docs site? hahahha

Comment (spydon) [reply .github/pull_request_template.md:20]: Aaah, I misunderstood Renan in the chat. But aren't we publishing docs manually to the site today anyways, why do we need the docs site before we remove develop?

Comment (erickzanardo) [reply .github/pull_request_template.md:20]: Many people still uses the docs on the Github, that is why I think Renan still wants to keep the develop.

IMO, that could be solved with a disclaimer on our mark down, saying that the current version docs is on the site...

Comment (spydon) [reply .github/pull_request_template.md:20]: Updated the Readme with links to the different doc versions, if we push a `latest` tag later we won't have to update the links when we do releases. Maybe we can do that once we release the first final v1.0.0 version.

File: README.md [hunk 2]
```diff
@@ -18,6 +18,21 @@ The current v1 release is `1.0.0-rc2` on pub. The last stable version so far is
 
 A minimalistic Flutter game engine.
 
+## Documentation
+
+Note that the documentation in the master branch of this repo is newer than the latest released version.
+
+Here you can find the documentation for different versions:
+- Latest stable version (0.28.x): [Flame-engine website](https://flame-engine.org/)
+- Latest [0.28.x release](https://github.com/flame-engine/flame/tree/master-v0.x/doc) GitHub docs
```
Comment (erickzanardo) [inline README.md:27]: This could be a hassle to maintain as we release more version, I think we could just omit the versions, what do you think?

**Review by erickzanardo — APPROVED** [review]
one nit, but LGTM already

File: README.md (hunk 2 above)
Comment (spydon) [reply README.md:27]: Good point, removing them!

**Review by renancaraujo — APPROVED** [review]

**Pull request merged**